Automattic/jetpack
# Full Sync :: Reduce Concurrency.

When the Full Sync logic was migrated from queue based to immediate sending of data there was a send lock implemented with an expiration of 15 seconds. This expiration of the lock allows for concurrent sends when the original request is still being processed. These concurrent requests can build up causing the reduction of application processes available to handle new requests to the site.

Fixes #
https://wp.me/p9dueE-2kk

#### Changes proposed in this Pull Request:
* Reduce concurrent sending of data during a Full Sync.

#### Does this pull request change what data or activity we track or use?
NO

#### Testing instructions:

* Apply Patch
* on a WP.com sandbox run `wp jetpack-sync pull --blog-id=XXXXX --start`
* Verify the Full Sync starts and completes.

#### Proposed changelog entry for your changes:
* Sync Performance Update : Reduce concurrency of sync requests to minimize number of application processes that are waiting for sync processing to complete.
